### PR TITLE
KIALI-1906 Support for security and non security environments

### PR DESCRIPTION
**Pull request by aljesusg**
With this we support the integration of Jaeger with Iframe when:

- kiali http and Jaeger http
- kiali https and Jaeger https

When one of them is with security and the other without we'll have the link to the jaeger website.

What do you think @lucasponce, @jotak ?

With this we only need to work in the JAEGER_URL provided by the backend or Makefile

### CODE REVIEW

**Comment by pilhuhn** [comment]
Sounds fair.

**Comment by aljesusg** [comment]
Hey @jotak I did the changes requested in the jaeger integration PR, can you review it?

File: src/config/paths.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/**
+ * Paths
+ */
+
+export enum paths {
```
Comment (jotak) [inline src/config/paths.ts]: Oops, wrong one, enums are PascalCase ;-)
https://github.com/basarat/typescript-book/blob/master/docs/styleguide/styleguide.md#enum

I think this file was good as it was before, also regarding its name (Path.ts) and the enum values (for which we decided to keep upper case)

File: src/pages/Overview/OverviewStatus.tsx [hunk 2]
```diff
@@ -12,7 +11,7 @@ type Props = {
   namespace: string;
   status: Status;
   items: string[];
-  targetPage: Paths;
+  targetPage: string;
```
Comment (jotak) [inline src/pages/Overview/OverviewStatus.tsx]: Why remove `Paths`? It's more specific typing, so better.

**Comment by jotak** [comment]
@aljesusg there's just a missing comment from the previous PR, not addressed yet:

https://github.com/kiali/kiali-ui/pull/965/files#r258911600 : we don't use `^` for semver in all other modules

**Comment by aljesusg** [comment]
> @aljesusg there's just a missing comment from the previous PR, not addressed yet:
> 
> https://github.com/kiali/kiali-ui/pull/965/files#r258911600 : we don't use `^` for semver in all other modules

Changed Thanks

**Comment by jotak** [comment]
So I tested these scenarios:
- Kiali on https and Jaeger on http: the tab opens Jaeger UI in a new window
- Kiali and Jaeger both on https: the tab shows Jaeger embedded

So I think this is what's expected.
The only suggestion I have would be to add an "external link" icon in the case it's opened in a new window (fa-external-link), so that there's no surprise for the user.

**Comment by aljesusg** [comment]
Done @jotak 
external_link

My concern is about the link of Distributed Tracing Link that opens jaeger in a new tab too

**Comment by jotak** [comment]
@aljesusg yes there's the same problem with the distributed tracing link. But at least, it doesn't affect the nominal case where both jaeger and kiali are under https. I would say it's good enough.

**Comment by hhovsepy** [comment]
When refreshing the Service Details page, it first appears "Traces" tab with external link, then within an 1/2 sec, switches to "Error Traces" tab.

**Comment by aljesusg** [comment]
This is by the redux state I think let me check it @hhovsepy

**Comment by aljesusg** [comment]
@hhovsepy I'll fix this problems in https://github.com/kiali/kiali-ui/pull/1025 see comment https://github.com/kiali/kiali-ui/pull/1025#issuecomment-467769657 Thanks

**Pull request merged**